src/v/redpanda/admin_server.cc

```diff
@@ -676,6 +692,49 @@ void admin_server::register_cluster_config_routes() {
               update.upsert.push_back({i.name.GetString(), s});
           }
 
+          // Config property validation happens further down the line
+          // at the point that properties are set on each node in
+          // response to the deltas that we write to the controller log,
+          // but we also do an early validation pass here to avoid writing
+          // clearly wrong things into the log & give better feedback
+          // to the API consumer.
+          if (!get_boolean_query_param(*req, "force")) {
+              // A scratch copy of configuration: we must not touch
+              // the real live configuration object, that will be updated
+              // by config_manager much after config is written to controller
+              // log.
+              config::configuration cfg;
+
+              std::map<ss::sstring, ss::sstring> errors;
+              for (const auto& i : update.upsert) {
+                  // Decode to a YAML object because that's what the property
+                  // interface expects.
+                  // Don't both catching ParserException: this was encoded
+                  // just a few lines above.
+                  const auto& yaml_value = i.second;
+                  auto val = YAML::Load(yaml_value);
+
+                  if (!cfg.contains(i.first)) {
+                      errors[i.first] = "Unknown property";
+                  }
+                  auto& property = cfg.get(i.first);
```

Comment:
did you intend to `continue` if the property wasn't found?

Reply:
I did not!  Good catch, adding a test permutation for it too.